x-pack/plugins/observability_onboarding/server/lib/get_authentication_api_key.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core-http-server';
+import { HTTPAuthorizationHeader } from '@kbn/security-plugin/server';
+
+export const getAuthenticationAPIKey = (request: KibanaRequest) => {
+  const authorizationHeader = HTTPAuthorizationHeader.parseFromRequest(request);
+  if (authorizationHeader && authorizationHeader.credentials) {
+    const apiKey = Buffer.from(authorizationHeader.credentials, 'base64')
+      .toString()
+      .split(':');
+    return {
+      apiKeyId: apiKey[0],
+      apiKey: apiKey[1],
```

Comment:
Have we discussed this approach with the Kibana security folks to be sure it's blessed? Also: do we have API tests to cover this if platform decides to change anything in this area?

Reply:
No we haven't discussed it since earlier today, but it's worth getting some clarification. Currently not covered by tests.

Reply:
Okay, we should get API test coverage for all of this before shipping.

Reply:
Issue for API tests: https://github.com/elastic/kibana/issues/159451

Reply:
Thank you!